scripts/airdropBsxToHdx/getVestingSchedules.ts

```diff
@@ -0,0 +1,85 @@
+// ✅  -> calculate BSX reward for accounts from .json of getData
+
+// ✅ -> calulate 5 000 000 000 proportional to holdings at the time for each holder
+//  ->  output a .json schedule
+
+// ✅  -> use same schedule as we used for 70% of each contributor's BSX crowdloan rewards
+
+const BN = require("BN.js")
+const airdropData = require("./data/airdropData.json")
+
+type DynamicVestingInfo = {
+    destination: string,
+    schedule: {
+        amountToBeVested: string,
+        start: string,
+        period: string,
+        per_period: string,
+        period_count: string 
+    }
+}
+
+type VestingBatch = DynamicVestingInfo[]
+
+const airdropBsxAllocation = new BN('5000000000000000000000')
+// https://basiliskfi.substack.com/p/introducing-basilisk 
+// ^^^ 25% of HDX holders ( HDX LBP participants, team and investors ) allocation
+const sumHdxBalances = new BN(airdropData.totalAllOgHdxBalances) 
+const bsxAirdroppedPerOneHdx = 
```

Comment:
this yields result `10`, but correctly it should be `10.987506461393665171570514346350267793394035843157015022354629124..` and that mean you will have some leftover tokens on your hands :)

Reply:
you have to always mul by 10^12, then divide, mul and finally divide by 10^12

Reply:
Thanks @lumir-mrkva.  I need to always double check my work. 

in the end I pulled in `bignumber.js` to accomplish this math,  because `bignumber.js` is more friendly with decimals ( for me.. ) 

thanks for educating me,  please do let me know what else can be done better